var/spack/repos/builtin/packages/py-pycompadre/package.py

```diff
@@ -35,19 +38,30 @@ class PyPycompadre(PythonPackage):
         description="Debugging level 0) release 1) debug 2) extreme debugging",
     )
 
-    depends_on("cmake@3.10.0:", type="build")
-    depends_on("python@3.4:", type=("build", "link", "run"))
-    depends_on("py-pip", type=("build", "link", "run"))
+    depends_on("cmake@3.13:", type="build", when="@:1.4")
```

Comment:
FYI. `cmake@3.10:` is required for `1.3.0`, `1.4.1`,  and `1.5.0` per `setup.py`s.

Reply:
Dependency on `@3.16:` doesn't show up until `1.6.0`.

Reply:
Thank you. All of those versions up to 1.4.1 had a requirement of 3.10.0 in setup.py. In 1.5.0 this tightened to `cmake@3.16:` in CMakeLists.txt, but remained 3.10 in setup.py only because it wasn't kept in sync with the requirement in the CMakeLists.txt that it invokes. CMakeLists.txt's tighter version requirements are enforced when the build script calls CMake. They should have been held together (setup.py and CMakeLists.txt), but if they differ then the tighter of the two should determine the requirement. I'll modify this for `@:1.4`.